### PR TITLE
Us123809/tic chart perf

### PR DESCRIPTION
**Pull request by johngwilkinson**
Quality Notes
- functional: demo page & e2e sanity check
- perf (w. 50k users in one course demo data):
  - before: 2.8s in chart.js (across all charts)
  - after: ~200ms in chart.js (across all charts), ~2ms in tic render
- devops/security: n/a
- ux
  - screen-reader, tooltips, and filtering work per-quadrant and are not affected
  - scatter chart shows fewer, quantized dots if there are over 1000 data points
  - this should not affect the usefulness of the chart as it still indicates roughly how data is distributed and works as a filter
  - [x] get UX approval

### CODE REVIEW

File: components/time-in-content-vs-grade-card.js [hunk 1]
```diff
@@ -66,15 +66,41 @@ export class TimeInContentVsGradeFilter {
...
+		let [lastTic, lastGrade] = sorted[0];
+		const grouped = [sorted[0]];
+		for (let i = 1; i < sorted.length; i++) {
+			const [tic, grade] = sorted[i];
+			if (tic === lastTic && grade === lastGrade) continue;
+			grouped.push(sorted[i]);
+			[lastTic, lastGrade] = sorted[i];
+		}
+
+		return { data: grouped, size: fullData.length };
```
Comment (bemailloux) [inline components/time-in-content-vs-grade-card.js:103]: What does the tooltip say when you've done the coalescing (as opposed to when all the data points are shown)? Is there a difference?

**Review by bemailloux — COMMENTED** [review]
Has this been demoed?

File: components/time-in-content-vs-grade-card.js (hunk 1 above)
Comment (johngwilkinson) [reply components/time-in-content-vs-grade-card.js:103]: No difference

**Comment by johngwilkinson** [comment]
> Has this been demoed?

No. I should run it by Kevin; todo added.

**Review by Vitalii-Misechko — APPROVED** [review]
LGTM!

**Review by bemailloux — APPROVED** [review]

**Pull request merged**